docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+   on `remainingInformation`.
+   The setDate function will then use the following regex to match.
+    - `-date\\s+(\\d{2}/\\d{2}/\\d{4})` matches the corresponding
+      structure`[-date] [whitespace] [date in DD/MM/YYYY format]`
+
+These steps will ultimately parse the user's input and extract the necessary information. If any pattern does not match,
+our parser will throw a `RainDayException` indicating the wrong input format.
+
+#### Alternatives considered
+
+We have tried using the .split() function of strings to extract information. However, as we introduced more features and
+the commands grew more complex, the .split() function became very messy and inconvenient. Using this will also result in
+us needing to do more exceptions and error handling, which just made the entire process very complicated. Thus, we opted
+to use regular expressions, which is a more tidy and logical way to parse the inputs.
+
+### Viewing your data `view`
+
+- The command `view` is used to view all statements, and a ViewCommand object will be created.
+    - The format for the command is `view TIMESPAN -sort`
+    - Any other characters after any valid view command are automatically ignored
+- Information is presented in a table format to help improve clarity for users.
+    - The table includes information in the summary such as whether it is sorted and the amount of history shown. 
+    - This was deliberate as presenting this information at the bottom makes it easier for users to spot, 
+      as placing them at the top may cause users to miss it if they have a large table
+
+![ViewCommandSequenceDiagram.png](images\DeveloperGuide\ViewCommandSequenceDiagram.png)
+
+### Design considerations
+- The limit for the timespans are deliberately set to cover common timespans
+    - One can view up to 31 days / 4 weeks, as they each make up a month
+    - Similarly, one can view up to 12 months, as they make up a year
+    - In line with the average user of the target audience, the limit for 10 years was set as a soft limit
+- The setting of hiding the value of transactions that are ignored is deliberate, as it is the most prominent and
+  direct way for users to see this.
+    - In line with the purpose of the Ignore Command, -all flag will also not show this. Instead, users can view the
+      value via the export command, as elaborated below
+
+### Setting your monthly Budget Goal `setbudget`
+
+- The command `setbudget` is used to set the user's monthly budget goal
+    - Format: `setbudget VALUE` 
+    - Users can also turn off the feature at any time by setting `VALUE` to 0
+- Once a goal is present, user's will be reminded of how close they are to sticking to their budget, or how
+  much they have exceeded it by.
+- This can be seen at start-up and when the user makes any changes to their expenses for the month.
+
+![SetBudgetCommandSequenceDiagram.png](images\DeveloperGuide\SetBudgetCommandSequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/40513040/229009668-d9b7979a-8033-407d-860b-312e9b446a48.png)

The `:UserData` activation bar should only start after the `setBudgetGoal` arrow